src/Twig/AssetRuntime.php

```diff
@@ -197,6 +197,42 @@ public function assetAverageColor(string $hash): string
         }
     }
 
+    /**
+     * @return array<string, int|string>|null
+     */
+    public function imageInfo(string $hash): ?array
+    {
+        $tempFile = $this->temporaryFile($hash);
+
+        if (null === $tempFile) {
+            return null;
+        }
+
+        $imageSize = \getimagesize($tempFile);
+
+        if (false === $imageSize) {
+            return null;
+        }
+
+        $imageInfo = [
+            'width' => $imageSize[0],
+            'height' => $imageSize[1],
+            'mimeType' => $imageSize['mime'],
+            'extension' => \explode('/', $imageSize['mime'])[1]
+        ];
+        
+        $imageResolution = \imageresolution(\imagecreatefromstring($this->storageManager->getContents($hash)));
```

Comment:
```suggestion
        $image = \imagecreatefromstring($this->storageManager->getContents($hash));
        if (false === $image) {
             throw new RuntimeException('Unexpected false image');
        }
        $imageResolution = \imageresolution($image);
```

Reply:
Maubert replies on this error that's a 0/0 image?

Reply:
Sorry I don't get that comment

Reply:
I was on my mobile phone. Instead of throwing a exception if the asset passed is not am image. Maybe we can considered to return meta about a 0 by 0 pixel information